api/src/main/java/io/opentelemetry/trace/TraceId.java

```diff
@@ -193,32 +163,7 @@ public boolean equals(@Nullable Object obj) {
    *
    * @return the rightmost 8 bytes of the trace-id as a long value.
    */
-  public long getTraceRandomPart() {
-    return idLo;
-  }
-
-  @Override
-  public int hashCode() {
-    // Copied from Arrays.hashCode(long[])
-    int result = 1;
-    result = 31 * result + ((int) (idHi ^ (idHi >>> 32)));
-    result = 31 * result + ((int) (idLo ^ (idLo >>> 32)));
-    return result;
-  }
-
-  @Override
-  public String toString() {
-    return "TraceId{traceId=" + toLowerBase16() + "}";
-  }
-
-  @Override
-  public int compareTo(TraceId that) {
-    if (idHi == that.idHi) {
-      if (idLo == that.idLo) {
-        return 0;
-      }
-      return idLo < that.idLo ? -1 : 1;
-    }
-    return idHi < that.idHi ? -1 : 1;
+  public static long getTraceIdRandomPart(CharSequence traceId) {
```

Comment:
```suggestion
  public static long getRandomPart(CharSequence traceId) {
```

We don't repeat the classname in the others, don't think we need to here either.

Reply:
Though this seems like it doesn't need to be public API, it even documents itself as being used in one specific sampler?

Reply:
This was the method name before, so I kept it as-is.  And, it's public so any Sampler could get access to it, I believe.